Modules/System/File Access/src/Connector/FileConnector.Interface.al

```diff
@@ -0,0 +1,144 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+/// <summary>
+/// An file connector interface used to creating file accounts and handle external files.
+/// </summary>
+interface "File Connector"
+{
+    /// <summary>
+    /// Gets a List of Files stored on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to get the file.</param>
+    /// <param name="Path">The file path to list.</param>
+    /// <param name="Files">A list with all files stored in the path.</param>
+    procedure ListFiles(AccountId: Guid; Path: Text; var FileAccountContent: Record "File Account Content" temporary);
```

Comment:
I'm not sure if we might need an additional optional parameter codeunit.
e.g. we might need to show only the first 100 files for performance reasons instead  of loading all 10.000 files in one call.
I've seen there some pagination for the sharepoint module.

Reply:
Good point. I will look if i can implement a generic pattern.